packages/Search/test/tstBoostGeometryAdapters.cpp

```diff
@@ -0,0 +1,248 @@
+    TEST_ASSERT( !bg::equals( point, bg::make<bg::Point>( 0., 0., 0. ) ) );
+
+    dtk::Box box = {{{1., 2., 3.}}, {{4., 5., 6.}}};
+    TEST_ASSERT( dtk::equals( box, {{{1., 2., 3.}}, {{4., 5., 6.}}} ) );
+    TEST_ASSERT( !dtk::equals( box, {{{0., 0., 0.}}, {{1., 1., 1.}}} ) );
+    TEST_ASSERT( bg::equals( box, dtk::Box{{{1., 2., 3.}}, {{4., 5., 6.}}} ) );
+    TEST_ASSERT( !bg::equals( box, dtk::Box{{{0., 0., 0.}}, {{1., 1., 1.}}} ) );
+    TEST_ASSERT(
+        bg::equals( box, bg::Box( bg::make<bg::Point>( 1., 2., 3. ),
+                                  bg::make<bg::Point>( 4., 5., 6. ) ) ) );
+    TEST_ASSERT(
+        !bg::equals( box, bg::Box( bg::make<bg::Point>( 0., 0., 0. ),
+                                   bg::make<bg::Point>( 1., 1., 1. ) ) ) );
+    // Now just for fun compare the DTK box to a Boost.Geometry box composed of
+    // DTK points.
+    TEST_ASSERT( bg::equals(
+        box, bg::model::box<dtk::Point>( {{1., 2., 3.}}, {{4., 5., 6.}} ) ) );
+    TEST_ASSERT( !bg::equals(
+        box, bg::model::box<dtk::Point>( {{0., 0., 0.}}, {{0., 0., 0.}} ) ) );
+}
+
+TEUCHOS_UNIT_TEST( BoostGeometryAdapters, distance )
+{
+    // NOTE unsure if should test for floating point equality here
+    dtk::Point a = {{0., 0., 0.}};
+    dtk::Point b = {{0., 1., 0.}};
+    TEST_EQUALITY( dtk::distance( a, b ), 1. );
+    TEST_EQUALITY( bg::distance( a, b ), 1. );
+
+    std::tie( a, b ) = std::make_pair<dtk::Point, dtk::Point>( {{0., 0., 0.}},
+                                                               {{1., 1., 1.}} );
+    TEST_EQUALITY( dtk::distance( a, b ), std::sqrt( 3. ) );
+    TEST_EQUALITY( bg::distance( a, b ), std::sqrt( 3. ) );
+
+    TEST_EQUALITY( dtk::distance( a, a ), 0. );
+    TEST_EQUALITY( bg::distance( a, a ), 0. );
+
+    dtk::Box unit_box = {{{0., 0., 0.}}, {{1., 1., 1.}}};
+    dtk::Point p = {{.5, .5, .5}};
+    // NOTE DTK has no overload distance( Box, Point )
+    TEST_EQUALITY( dtk::distance( p, unit_box ), 0. );
+    // TEST_EQUALITY( dtk::distance( unit_box, p ), 0. );
+    TEST_EQUALITY( bg::distance( p, unit_box ), 0. );
+    TEST_EQUALITY( bg::distance( unit_box, p ), 0. );
+
+    p = {{-1., -1., -1.}};
+    TEST_EQUALITY( dtk::distance( p, unit_box ), std::sqrt( 3. ) );
+    TEST_EQUALITY( bg::distance( p, unit_box ), std::sqrt( 3. ) );
+
+    p = {{-1., .5, -1.}};
+    TEST_EQUALITY( dtk::distance( p, unit_box ), std::sqrt( 2. ) );
+    TEST_EQUALITY( bg::distance( p, unit_box ), std::sqrt( 2. ) );
+
+    p = {{-1., .5, .5}};
+    TEST_EQUALITY( dtk::distance( p, unit_box ), 1. );
+    TEST_EQUALITY( bg::distance( p, unit_box ), 1. );
+}
+
+TEUCHOS_UNIT_TEST( BoostGeometryAdapters, expand )
+{
+    using dtk::equals;
+    dtk::Box box;
+    // NOTE even though not considered valid, default constructed DTK box can be
+    // expanded using Boost.Geometry algorithm.
+    TEST_ASSERT( !bg::is_valid( box ) );
+    bg::expand( box, dtk::Point{{0., 0., 0.}} );
+    dtk::expand( box, {{1., 1., 1.}} );
```

Comment:
so these `expand` functions expand the box to encapsulate the point?

Reply:
Correct.